unstructured/documents/html.py

```diff
@@ -441,6 +456,36 @@ def _is_text_tag(tag_elem: etree.Element, max_predecessor_len: int = 5) -> bool:
     return False
 
 
+def _process_leaf_table_item(
+    tag_elem: etree.Element,
+) -> Tuple[Optional[Element], etree.Element]:
+    if tag_elem.tag in TABLE_TAGS:
+        nested_table = tag_elem.findall("table")
+        if not nested_table:
+            rows = tag_elem.findall("tr")
+            if not rows:
+                body = tag_elem.find("tbody")
+                rows = body.findall("tr")
+            if len(rows) > 0:
+                table_data = [list(row.itertext()) for row in rows]
+                html_table = tabulate(table_data, tablefmt="html")
+                table_text = " ".join(" ".join(row) for row in table_data).strip()
+            else:
+                table_text = ""
+                html_table = ""
+            return (
+                HTMLTable(
+                    text=table_text,
+                    text_as_html=html_table.replace("\n", "<br>"),
+                    tag=tag_elem.tag,
+                    ancestortags=tuple(el.tag for el in tag_elem.iterancestors())[::-1],
+                ),
+                tag_elem,
```

Comment:
why does `next_element` need to be returned? In the case of `_process_list_item` the docstring mentions "Also returns the next html elements so that we can skip processing if bullets are found in a div element.", but I don't see a similar use case here. Can remove returning `next_element` if not needed or used (which it doesn't look like it is where it's being called in the case in `class HTMLDocument`), or add a comment on why it's necessary to keep track of.